### ISSUE TITLE
OpenJDK fails

### ISSUE THREAD

**Issue by nmaguiar** [tbodt/ish#306]
After installing openjdk8 and trying to run it fails:
````sh
iPad:~# java
OpenJDK Server VM warning: Can't detect initial thread stack location - find_vma failed           
Error occurred during initialization of VM       
Too small initial heap     
````
On dmesg there is a FIXME message that is probably the first reason why it doesn’t work:
````
FIXME no support for shared futexes
````
It would be amazing if one day it works partially or completely :)

**Comment by tbodt** [comment]
Sounds a little like missing /proc/pid/maps

**Comment by Teg79** [comment]
I have the same problem, it will be great to have the jvm running

**Comment by PseudonymPatel** [comment]
So I've tried for a while to get any forms of java running, and I was playing around, and I tried these two commands:  
![IMG_0994](https://user-images.githubusercontent.com/20823991/56459016-0f849900-635c-11e9-93fd-e3ec816977f0.jpeg)  
Here are the two error logs:  
[hs_err_pid40.txt](https://github.com/tbodt/ish/files/3100268/hs_err_pid40.txt)
[hs_err_pid43.txt](https://github.com/tbodt/ish/files/3100269/hs_err_pid43.txt)

Here are my thoughts, but I could be wrong because I don't have much experience with this:  
it seems that running in mixed mode is not the problem, because running in the interactive mode is not working either. The files show that the machine points to `LD_LIBRARY_PATH=/usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/server:/usr/lib/jvm/java-1.8-openjdk/jre/lib/i386:/usr/lib/jvm/java-1.8-openjdk/jre/../lib/i386` but ish is `i686`. I'm hoping for someone with more experience could help.

<hr/>
On a side note, I've also tried installing 32bit and 64bit tar.gz java. I tried installing Java 1.8.0_201, here is my release file:
[release.txt](https://github.com/tbodt/ish/files/3100288/release.txt)
My error that I get when I run any command (`javac`, `java`, `javadoc`, etc.) is:  
`./java: line 1: syntax error: unexpected word (expecting ")")`

**Comment by craigotis** [comment]
Note that Java 7 seems to work, with `api add openjdk7`, you can run:
```
iPad:~# java -mx256m -version                                                                                                              
OpenJDK Client VM warning: Can't detect initial thread stack location - find_vma failed                                                    
java version "1.7.0_211"                                                                                                                   
OpenJDK Runtime Environment (IcedTea 2.6.17) (Alpine 7.211.2.6.17-r0)                                                                      
OpenJDK Client VM (build 24.211-b02, mixed mode)                                                                                           
iPad:~#  
```
But I can't get Java 8 to work for the same issues you're encountering. Further, if I set `-mx512m`, then the Java process spits out the version, and a _split_ second later, Ish crashes.

**Comment by craigotis** [comment]
Does anyone know how to force `apk` to install a package without its dependencies? I was hoping to play around with Maven, but it has a dependency on `openjdk-jre8`, which seems to work less well than Java 7.

**Comment by tbodt** [comment]
I don't think that's possible with apk. There are other ways to install maven though.

**Comment by Teg79** [comment]
You can download maven in any folder, unpack it and run it, be sure to
download a version compatible with jdk 7

Il giorno lun 24 giu 2019 alle ore 22:59 Theodore Dubois <
notifications@github.com> ha scritto:

**Comment by craigotis** [comment]
Yep - I was hoping to maintain an `apk` script for easy reproducibility, but just grabbed the Maven binary from their website and ran it with OpenJDK 7. It failed pretty quickly (there’s more, but it gets less helpful):
```
iPad:~/apps/apache-maven-3.6.1/bin# ./mvn                                                                                                                          
OpenJDK Client VM warning: Can't detect initial thread stack location - find_vma failed                                                                            
#                                                                                                                                                                  
# A fatal error has been detected by the Java Runtime Environment:                                                                                                 
#                                                                                                                                                                  
#  SIGILL (0x4) at pc=0x00000000, pid=14, tid=4149333376                                                                                                           
#                                                                                                                                                                  
# JRE version: OpenJDK Runtime Environment (7.0_211-b02) (build 1.7.0_211-b02)                                                                                     
# Java VM: OpenJDK Client VM (24.211-b02 mixed mode linux-x86 )                                                                                                    
# Derivative: IcedTea 2.6.17                                                                                                                                       
# Distribution: Custom build (Thu Apr 11 13:17:01 UTC 2019)                                                                                                        
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again                           
#                                                                                                                                                                  
# An error report file with more information is saved as:                                                                                                          
# /root/apps/apache-maven-3.6.1/bin/hs_err_pid14.log                                                                                                               
#                                                                                                                                                                  
# A fatal error has been detected by the Java Runtime Environment:                                                                                                 
#                                                                                                                                                                  
#  SIGILL (0x4) at pc=0x00000000, pid=44, tid=4149333376                                                                                                           
#                                                                                                                                                                  
# JRE version: OpenJDK Runtime Environment (7.0_211-b02) (build 1.7.0_211-b02)                                                                                     
# Java VM: OpenJDK Client VM (24.211-b02 mixed mode linux-x86 )                                                                                                    
# Derivative: IcedTea 2.6.17                                                                                                                                       
# Distribution: Custom build (Thu Apr 11 13:17:01 UTC 2019)                                                                                                        
# Core dump written. Default location: /root/apps/apache-maven-3.6.1/bin/core or core.44                                                                           
#                                                                                                                                                                  
# If you would like to submit a bug report, please include                                                                                                         
# instructions on how to reproduce the bug and visit:                                                                                                              
#   http://icedtea.classpath.org/bugzilla                                                                                                                          
#                                                                                                                                                                  
                                                                                                                                                                   
---------------  T H R E A D  ---------------                                                                                                                      
                                                                                                                                                                   
Current thread (0x56561400):  JavaThread "main" [_thread_in_Java, id=61, stack(0xf74cc000,0xf751cd60)]                                                             
                                                                                                                                                                   
Stack: [0xf74cc000,0xf751cd60]                                                                                                                                     
[error occurred during error reporting (printing stack bounds), id 0xb]
```
I’m guessing the SIGKILL is coming from iOS. Is it memory-related? Note also I have `$MAVEN_OPTS` set to `-Xmx384m` but most other values (256/512) fail the same way.

**Comment by tbodt** [comment]
That's a SIGILL which is very different from a SIGKILL. dmesg will tell you more, but in all likelihood it's missing sse instructions

**Comment by craigotis** [comment]
I cleared `dmesg` then ran it immediately after a Maven failure:
```
iPad:~/apps/apache-maven-3.6.1/bin# ./mvn                                                                                                                          
OpenJDK Client VM warning: Can't detect initial thread stack location - find_vma failed                                                                            
#                                                                                                                                                                  
# A fatal error has been detected by the Java Runtime Environment:                                                                                                 
#                                                                                                                                                                  
#  SIGILL (0x4) at pc=0x00000000, pid=85, tid=4149333376                                                                                                           
#                                                                                                                                                                  
# JRE version: OpenJDK Runtime Environment (7.0_211-b02) (build 1.7.0_211-b02)                                                                                     
# Java VM: OpenJDK Client VM (24.211-b02 mixed mode linux-x86 )                                                                                                    
# Derivative: IcedTea 2.6.17                                                                                                                                       
# Distribution: Custom build (Thu Apr 11 13:17:01 UTC 2019)                                                                                                        
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again                           
#                                                                                                                                                                  
# An error report file with more information is saved as:                                                                                                          
# /root/apps/apache-maven-3.6.1/bin/hs_err_pid85.log                                                                                                               
#                                                                                                                                                                  
# If you would like to submit a bug report, please include                                                                                                         
# instructions on how to reproduce the bug and visit:                                                                                                              
#   http://icedtea.classpath.org/bugzilla                                                                                                                          
#                                                                                                                                                                  
Aborted                                                                                                                                                            
iPad:~/apps/apache-maven-3.6.1/bin# dmesg                                                                                                                          
unknown ioctl 5413                                                                                                                                                 
unknown ioctl 5413                                                                                                                                                 
unknown ioctl 5413                                                                                                                                                 
unknown ioctl 5413                                                                                                                                                 
unknown ioctl 5413                                                                                                                                                 
unknown ioctl 5413                                                                                                                                                 
unknown ioctl 5413                                                                                                                                                 
unknown ioctl 5413                                                                                                                                                 
unknown ioctl 5413                                                                                                                                                 
unknown ioctl 5413                                                                                                                                                 
FIXME no support for shared futexes                                                                                                                                
102 illegal instruction at 0xf548a247: dd c7 dd c6 dd c5 dd c4                                                                                                     
102 page fault on 0x4 at 0xf788452a                                                                                                                                
102 page fault on 0x4 at 0xf788452a
```

**Comment by tbodt** [comment]
ffree

**Comment by tbodt** [comment]
Just fixed the warning about not being able to detect initial stack location. Everything else is still broken.

**Comment by tbodt** [comment]
Now tries to use cmpxchg8b

**Comment by Nikita11790** [comment]
I got java to work after running `reboot` And letting the app crash, once I opened it, it worked

**Comment by jsjohnst** [comment]
Just an update on this thread, but using the AppStore version of iSH and OpenJDK8 I was able to compile a quick HelloWorld Java example and execute it successfully (albeit I needed to specify `-ms128m` to avoid the heap size error previously mentioned, 256/512/1024m also work).

**Comment by damesek** [comment]
That is a great news! I will check it! Thanks the info :)

**Comment by maxandersen** [comment]
noob question - whats the command to run to install openjdk8?

**Comment by sujansa** [comment]
After some trial and error, I'm able to get it to work on my (embarrassingly old) iPad. Although both OpenJDK7 and OpenJDK8 worked, I had to play around with the heap size argument. Here's what I was able to do (in case anyone finds it useful).

# Install Java 7
apk add openjdk7 # It takes a while to complete installation

# This worked with openjdk7
java -ms128 -mx1g --version

# Remove Java 7
apk del openjdk7

apk add openjdk8

# This worked with openjdk8
java -ms128 -mx512m --version


I'm actually trying to see if there's a way to install Groovy. I'll keep you posted if I succeed but if anyone has any advice, you're most welcome.

**Comment by chchen2021** [comment]
As pointed out by @sujansa: the "too small initial heap" error can be resolved by increasing the heap size. The openjdk8 works smoothly with:
`java -mx512m`

It looks like only 32bit system has the heap size issue. On iSH, `-mx1024m` does not always work so `-mx512m` is probably the best option.

**Comment by zaratr** [comment]
> As pointed out by @sujansa: the "too small initial heap" error can be resolved by increasing the heap size. The openjdk8 works smoothly with: `java -mx512m`
> 
> It looks like only 32bit system has the heap size issue. On iSH, `-mx1024m` does not always work so `-mx512m` is probably the best option.

this works for me as well as the above thread mentions. however, not able to run compiler still. My files seem to be out of scope with an error "could not find or load main class ****.java"

-furter, I don't see javac as stated above. So I'm not sure how to compile programs with java. I have downloaded gdb successfully.

**Comment by razuuttarabd** [comment]
> Just an update on this thread, but using the AppStore version of iSH and OpenJDK8 I was able to compile a quick HelloWorld Java example and execute it successfully (albeit I needed to specify `-ms128m` to avoid the heap size error previously mentioned, 256/512/1024m also work).

According to you, I could run java using "java -ms128m -version". But how did you able to compile, cause 'javac -ms128m -test.java' throws me the error "-ash: javac: not found".

**Comment by craigotis** [comment]
@razuuttarabd Make sure you're installing the JDK, and not the JRE.

**Comment by maxandersen** [comment]
apk install openjdk8 does not seem to give the JDK but a JRE. 

What's the right apk install command?

**Comment by deepakHonakeri05** [comment]
@craigotis 
What is the correct app install command ?

**Comment by SageTTs** [comment]
> After some trial and error, I'm able to get it to work on my (embarrassingly old) iPad. Although both OpenJDK7 and OpenJDK8 worked, I had to play around with the heap size argument. Here's what I was able to do (just in case anyone finds it useful).
> 
> ```
> 
> # Install Java 7
> 
> apk add openjdk7 # It takes a while to complete installation
> 
> 
> 
> # This worked with openjdk7
> 
> java -ms128 -mx1g --version
> 
> 
> 
> # Remove Java 7
> 
> apk del openjdk7
> 
> 
> 
> apk add openjdk8
> 
> 
> 
> # This worked with openjdk8
> 
> java -ms128 -mx512m --version
> 
> ```
> 
> 
> 
> I'm actually trying to see if there's a way to install Groovy. I'll keep you posted if I succeed but if anyone has any advice, you're most welcome.

Hello I'm very new to all of this but after running above prompt I get this;
Error: Could not create the java virtual machine 
Error: A fatal exception has occurred the program will exit

**Comment by neppramod** [comment]
> > Just an update on this thread, but using the AppStore version of iSH and OpenJDK8 I was able to compile a quick HelloWorld Java example and execute it successfully (albeit I needed to specify `-ms128m` to avoid the heap size error previously mentioned, 256/512/1024m also work).
> 
> According to @jsjohnst, I could run java using "java -ms128m -version". But how did you able to compile, cause 'javac -ms128m test.java' or 'javac test.java' throws me the error "-ash: javac: not found".
> 
> Please help 🙏🏽

For compiling java program you can use "javac -J-ms128m Hello.java"

**Comment by maxandersen** [comment]
Which apk add brings javac?

Apk add openjdk8 only seem to bring java not javac?

**Comment by maxandersen** [comment]
Found how to run it but javac fails:


JAVA_HOME=/usr/lib/jvm/java-1.8-openjdk
PATH="$JAVA_HOME/bin:${PATH}"

**Comment by stantonk** [comment]
It segfaults for me:

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0xe80c4f30, pid=131, tid=0xf70afd90
#
# JRE version: OpenJDK Runtime Environment (8.0_302-b08) (build 1.8.0_302-b08)
# Java VM: OpenJDK Server VM (25.302-b08 mixed mode linux-x86 )
# Derivative: IcedTea 3.20.0
# Distribution: Custom build (Wed Dec  8 12:39:55 UTC 2021)
# Problematic frame:
# J 79 C1 sun.misc.MetaIndex.mayContain(Ljava/lang/String;)Z (51 bytes) @ 0xe80c4f30 [0xe80c4de0+0x150]
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# If you would like to submit a bug report, please include
# instructions on how to reproduce the bug and visit:
#   https://icedtea.classpath.org/bugzilla
#

---------------  T H R E A D  ---------------

Current thread (0xf7017c00):  JavaThread "main" [_thread_in_Java, id=132, stack(0xf705f000,0xf70afd70)]

siginfo: si_signo: 11 (SIGSEGV), si_code: 1 (SEGV_MAPERR), si_addr: 0x634c2108

Registers:
EAX=0xe80c4e00, EBX=0x634c2100, ECX=0xe76a3488, EDX=0xe743e9a0
ESP=0xf70ac3b0, EBP=0xf70ac4bc, ESI=0x00000025, EDI=0xe7449330
EIP=0xe80c4f30, EFLAGS=0x00000202, CR2=0x634c2108

Top of Stack: (sp=0xf70ac3b0)
0xf70ac3b0:   e7688ab8 00000000 56559468 f767aeb5
0xf70ac3c0:   56559468 00000000 00000000 00000000
0xf70ac3d0:   00000000 00000206 e76a3488 e76a3488
0xf70ac3e0:   e743e9a0 f70ac3e8 e71082f8 e743e9a0
0xf70ac3f0:   e76a3488 e80c4e00 f70ac4bc e80bbcc4
0xf70ac400:   f70ac418 f70ac408 e7149b80 00000000
0xf70ac410:   e7688ab8 e80b5c80 f70ac4bc e80bb304
0xf70ac420:   00000010 f73d88da f7af88d0 f766bde5 

Instructions: (pc=0xe80c4f30)
0xe80c4f10:   8f fb e6 81 e7 f8 ff 7f 00 83 ff 00 0f 84 26 03
0xe80c4f20:   00 00 8b 72 08 8b 76 08 bf 58 5f 67 e7 8b 5f 08
0xe80c4f30:   8b 5b 08 8b c2 89 7c 24 2c b9 18 26 1f e7 8b 40
0xe80c4f40:   04 3b 81 d0 00 00 00 75 0c 83 81 d4 00 00 00 01 

Register to memory mapping:

EAX=0xe80c4e00 is at entry_point+32 in (nmethod*)0xe80c4c88
EBX=0x634c2100 is an unknown value
ECX=0xe76a3488 is an oop
sun.misc.MetaIndex 
 - klass: 'sun/misc/MetaIndex'
EDX=0xe743e9a0 is an oop
java.lang.String 
 - klass: 'java/lang/String'
ESP=0xf70ac3b0 is pointing into the stack for thread: 0xf7017c00
EBP=0xf70ac4bc is pointing into the stack for thread: 0xf7017c00
ESI=0x00000025 is an unknown value
EDI=
[error occurred during error reporting (printing register info), id 0xb]

Stack: [0xf705f000,0xf70afd70],  sp=0xf70ac3b0,  free space=308k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
J 79 C1 sun.misc.MetaIndex.mayContain(Ljava/lang/String;)Z (51 bytes) @ 0xe80c4f30 [0xe80c4de0+0x150]
J 53 C1 sun.misc.URLClassPath$JarLoader.getResource(Ljava/lang/String;Z)Lsun/misc/Resource; (85 bytes) @ 0xe80bbcc4 [0xe80bbc00+0xc4]
j  sun.misc.URLClassPath.getResource(Ljava/lang/String;Z)Lsun/misc/Resource;+62
j  java.net.URLClassLoader$1.run()Ljava/lang/Class;+26
j  java.net.URLClassLoader$1.run()Ljava/lang/Object;+1
v  ~StubRoutines::call_stub
V  [libjvm.so+0x2f059c]
C  0xf7f5e8a0


---------------  P R O C E S S  ---------------

Java Threads: ( => current thread )
  0xe6971c00 JavaThread "Service Thread" daemon [_thread_blocked, id=141, stack(0xe686a000,0xe68bad70)]
  0xe6970400 JavaThread "C1 CompilerThread3" daemon [_thread_blocked, id=140, stack(0xe68ed000,0xe696dd70)]
  0xe6b96c00 JavaThread "C2 CompilerThread2" daemon [_thread_blocked, id=139, stack(0xe6999000,0xe6a19d70)]
  0xe6b96000 JavaThread "C2 CompilerThread1" daemon [_thread_blocked, id=138, stack(0xe6a3f000,0xe6abfd70)]
  0xe6b95800 JavaThread "C2 CompilerThread0" daemon [_thread_blocked, id=137, stack(0xe6ac1000,0xe6b41d70)]
  0xe6b94000 JavaThread "Signal Dispatcher" daemon [_thread_blocked, id=136, stack(0xe6b42000,0xe6b92d70)]
  0xe6c71c00 JavaThread "Finalizer" daemon [_thread_blocked, id=135, stack(0xe6bca000,0xe6c1ad70)]
  0xe6c71400 JavaThread "Reference Handler" daemon [_thread_blocked, id=134, stack(0xe6c1f000,0xe6c6fd70)]
=>0xf7017c00 JavaThread "main" [_thread_in_Java, id=132, stack(0xf705f000,0xf70afd70)]

Other Threads:
  0xe6cf7400 VMThread [stack: 0xe6c74000,0xe6cf4d70] [id=133]
  0xe6cf8000 WatcherThread [stack: 0xe67e9000,0xe6869d70] [id=142]

VM state:not at safepoint (normal execution)

VM Mutex/Monitor currently owned by a thread: None

Heap:
 def new generation   total 2432K, used 562K [0xe7400000, 0xe76a0000, 0xe76a0000)
  eden space 2176K,  14% used [0xe7400000, 0xe744cb58, 0xe7620000)
  from space 256K, 100% used [0xe7660000, 0xe76a0000, 0xe76a0000)
  to   space 256K,   0% used [0xe7620000, 0xe7620000, 0xe7660000)
 tenured generation   total 5504K, used 469K [0xe76a0000, 0xe7c00000, 0xe7c00000)
   the space 5504K,   8% used [0xe76a0000, 0xe7715768, 0xe7715800, 0xe7c00000)
 Metaspace       used 2778K, capacity 2788K, committed 2840K, reserved 4400K

Card table byte_map: [0xe7c2f000,0xe7c34000] byte_map_base: 0xe74f5000

Polling page: 0xf701e000

CodeCache: size=245760Kb used=873Kb max_used=873Kb free=244886Kb
 bounds [0xe7ffc000, 0xe823c000, 0xf6ffc000]
 total_blobs=283 nmethods=121 adapters=77
 compilation: enabled

Compilation events (10 events):
Event: 3.486 Thread 0xe6970400  115       3       java.util.zip.ZipFile::access$400 (5 bytes)
Event: 3.487 Thread 0xe6970400 nmethod 115 0xe80d31c8 code [0xe80d32e0, 0xe80d33a0]
Event: 3.571 Thread 0xe6b96000  116       4       java.lang.String::hashCode (55 bytes)
Event: 3.609 Thread 0xe6b96000 nmethod 116 0xe80d14c8 code [0xe80d15c0, 0xe80d16f0]
Event: 3.637 Thread 0xe6970400  119       3       java.util.jar.JarFile::maybeInstantiateVerifier (96 bytes)
Event: 3.650 Thread 0xe6970400 nmethod 119 0xe80d1a08 code [0xe80d1be0, 0xe80d2634]
Event: 3.657 Thread 0xe6970400  120       3       java.lang.String::regionMatches (149 bytes)
Event: 3.671 Thread 0xe6970400 nmethod 120 0xe80d4488 code [0xe80d46a0, 0xe80d5100]
Event: 3.751 Thread 0xe6970400  121       3       sun.net.www.ParseUtil::encodePath (336 bytes)
Event: 3.769 Thread 0xe6970400 nmethod 121 0xe80d5a48 code [0xe80d5c80, 0xe80d6918]

GC Heap History (2 events):
Event: 3.502 GC heap before
{Heap before GC invocations=0 (full 0):
 def new generation   total 2432K, used 2176K [0xe7400000, 0xe76a0000, 0xe76a0000)
  eden space 2176K, 100% used [0xe7400000, 0xe7620000, 0xe7620000)
  from space 256K,   0% used [0xe7620000, 0xe7620000, 0xe7660000)
  to   space 256K,   0% used [0xe7660000, 0xe7660000, 0xe76a0000)
 tenured generation   total 5504K, used 0K [0xe76a0000, 0xe7c00000, 0xe7c00000)
   the space 5504K,   0% used [0xe76a0000, 0xe76a0000, 0xe76a0200, 0xe7c00000)
 Metaspace       used 2693K, capacity 2724K, committed 2840K, reserved 4400K
Event: 3.545 GC heap after
Heap after GC invocations=1 (full 0):
 def new generation   total 2432K, used 256K [0xe7400000, 0xe76a0000, 0xe76a0000)
  eden space 2176K,   0% used [0xe7400000, 0xe7400000, 0xe7620000)
  from space 256K, 100% used [0xe7660000, 0xe76a0000, 0xe76a0000)
  to   space 256K,   0% used [0xe7620000, 0xe7620000, 0xe7660000)
 tenured generation   total 5504K, used 469K [0xe76a0000, 0xe7c00000, 0xe7c00000)
   the space 5504K,   8% used [0xe76a0000, 0xe7715768, 0xe7715800, 0xe7c00000)
 Metaspace       used 2693K, capacity 2724K, committed 2840K, reserved 4400K
}

Deoptimization events (0 events):
No events

Classes redefined (0 events):
No events

Internal exceptions (8 events):
Event: 0.558 Thread 0xf7017c00 Exception <a 'java/lang/NoSuchMethodError': Method sun.misc.Unsafe.defineClass(Ljava/lang/String;[BII)Ljava/lang/Class; name or signature does not match> (0xe7406eb0) thrown at [/home/buildozer/aports/community/openjdk8/src/icedtea-3.20.0/openjdk/hotspot/
Event: 0.559 Thread 0xf7017c00 Exception <a 'java/lang/NoSuchMethodError': Method sun.misc.Unsafe.prefetchRead(Ljava/lang/Object;J)V name or signature does not match> (0xe7407180) thrown at [/home/buildozer/aports/community/openjdk8/src/icedtea-3.20.0/openjdk/hotspot/src/share/vm/prims
Event: 3.060 Thread 0xf7017c00 Exception <a 'java/security/PrivilegedActionException'> (0xe759f0c8) thrown at [/home/buildozer/aports/community/openjdk8/src/icedtea-3.20.0/openjdk/hotspot/src/share/vm/prims/jvm.cpp, line 1520]
Event: 3.060 Thread 0xf7017c00 Exception <a 'java/security/PrivilegedActionException'> (0xe759f4a8) thrown at [/home/buildozer/aports/community/openjdk8/src/icedtea-3.20.0/openjdk/hotspot/src/share/vm/prims/jvm.cpp, line 1520]
Event: 3.093 Thread 0xf7017c00 Exception <a 'java/security/PrivilegedActionException'> (0xe75a1a50) thrown at [/home/buildozer/aports/community/openjdk8/src/icedtea-3.20.0/openjdk/hotspot/src/share/vm/prims/jvm.cpp, line 1520]
Event: 3.094 Thread 0xf7017c00 Exception <a 'java/security/PrivilegedActionException'> (0xe75a1e30) thrown at [/home/buildozer/aports/community/openjdk8/src/icedtea-3.20.0/openjdk/hotspot/src/share/vm/prims/jvm.cpp, line 1520]
Event: 3.209 Thread 0xf7017c00 Exception <a 'java/security/PrivilegedActionException'> (0xe75d8880) thrown at [/home/buildozer/aports/community/openjdk8/src/icedtea-3.20.0/openjdk/hotspot/src/share/vm/prims/jvm.cpp, line 1520]
Event: 3.210 Thread 0xf7017c00 Exception <a 'java/security/PrivilegedActionException'> (0xe75d8c60) thrown at [/home/buildozer/aports/community/openjdk8/src/icedtea-3.20.0/openjdk/hotspot/src/share/vm/prims/jvm.cpp, line 1520]

Events (10 events):
Event: 3.783 loading class com/sun/tools/javac/code/Symbol$MethodSymbol
Event: 3.783 loading class com/sun/tools/javac/code/Symbol$MethodSymbol done
Event: 3.803 loading class javax/lang/model/element/ExecutableElement
Event: 3.803 loading class javax/lang/model/element/Element
Event: 3.804 loading class javax/lang/model/element/Element done
Event: 3.804 loading class javax/lang/model/element/Parameterizable
Event: 3.805 loading class javax/lang/model/element/Parameterizable done
Event: 3.805 loading class javax/lang/model/element/ExecutableElement done
Event: 3.810 loading class com/sun/tools/javac/code/Symbol
Event: 3.810 loading class com/sun/tools/javac/code/Symbol done


Dynamic libraries:
56555000-56556000 r--p 00000000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/bin/javac
56556000-56557000 r--p 00001000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/bin/javac
56557000-56558000 r--p 00002000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/bin/javac
56558000-56559000 r--p 00002000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/bin/javac
56559000-5655a000 rw-p 00002000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/bin/javac
5655a000-5655b000 ---p 00000000 00:00 0          
5655b000-5655f000 -w-p 00000000 00:00 0          
e655f000-e6597000 rw-p 00000000 00:00 0          
e659b000-e65ba000 rw-p 00000000 00:00 0          
e65bb000-e66fb000 rw-p 00000000 00:00 0          
e66fb000-e66ff000 r--p 00000000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libnet.so
e66ff000-e6710000 r-xp 00004000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libnet.so
e6710000-e671a000 r--p 00015000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libnet.so
e671a000-e671b000 r--p 0001e000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libnet.so
e671b000-e671c000 rw-p 0001e000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libnet.so
e671c000-e6722000 r--p 00000000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libnio.so
e6722000-e672b000 r-xp 00006000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libnio.so
e672b000-e6731000 r--p 0000f000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libnio.so
e6731000-e6732000 r--p 00014000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libnio.so
e6732000-e6733000 rw-p 00014000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libnio.so
e6733000-e6767000 rw-p 00000000 00:00 0          
e6767000-e67e0000 r--- 007c1000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/lib/tools.jar
e67e0000-e67e8000 rw-p 00000000 00:00 0          
e67e8000-e67e9000 ---p 00000000 00:00 0          
e67e9000-e686a000 rw-p 00000000 00:00 0          
e686a000-e686d000 ---p 00000000 00:00 0          
e686d000-e68ec000 rw-p 00000000 00:00 0          
e68ec000-e68ed000 ---p 00000000 00:00 0          
e68ed000-e68f0000 ---p 00000000 00:00 0          
e68f0000-e696e000 rw-p 00000000 00:00 0          
e696e000-e6998000 rw-p 00000000 00:00 0          
e6998000-e6999000 ---p 00000000 00:00 0          
e6999000-e699c000 ---p 00000000 00:00 0          
e699c000-e6a1a000 rw-p 00000000 00:00 0          
e6a1a000-e6a3e000 rw-p 00000000 00:00 0          
e6a3e000-e6a3f000 ---p 00000000 00:00 0          
e6a3f000-e6a42000 ---p 00000000 00:00 0          
e6a42000-e6ac0000 rw-p 00000000 00:00 0          
e6ac0000-e6ac1000 ---p 00000000 00:00 0          
e6ac1000-e6ac4000 ---p 00000000 00:00 0          
e6ac4000-e6b42000 rw-p 00000000 00:00 0          
e6b42000-e6b45000 ---p 00000000 00:00 0          
e6b45000-e6b9e000 rw-p 00000000 00:00 0          
e6b9e000-e6ba5000 r--- 0006e000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/jfr.jar
e6ba5000-e6bca000 rw-p 00000000 00:00 0          
e6bca000-e6bcd000 ---p 00000000 00:00 0          
e6bcd000-e6c1f000 rw-p 00000000 00:00 0          
e6c1f000-e6c22000 ---p 00000000 00:00 0          
e6c22000-e6c73000 rw-p 00000000 00:00 0          
e6c73000-e6c74000 ---p 00000000 00:00 0          
e6c74000-e6cf5000 rw-p 00000000 00:00 0          
e6cf5000-e6d82000 rw-p 00000000 00:00 0          
e6d82000-e6f54000 r--- 01f92000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/rt.jar
e6f54000-e727a000 rw-p 00000000 00:00 0          
e727a000-e7400000 ---p 00000000 00:00 0          
e7400000-e7c45000 rw-p 00000000 00:00 0          
e7c45000-e7ffc000 ---p 00000000 00:00 0          
e7ffc000-e823c000 rwxp 00000000 00:00 0          
e823c000-f6ffc000 ---p 00000000 00:00 0          
f6ffc000-f6ffe000 rw-p 00000000 00:00 0          
f6ffe000-f7000000 r--p 00000000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libzip.so
f7000000-f7005000 r-xp 00002000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libzip.so
f7005000-f7008000 r--p 00007000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libzip.so
f7008000-f7009000 r--p 00009000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libzip.so
f7009000-f700a000 rw-p 00009000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libzip.so
f700a000-f700f000 rw-p 00000000 00:00 0          
f700f000-f7017000 rw-- 00000000 00:00 0          /tmp/hsperfdata_root/131
f7017000-f701d000 rw-p 00000000 00:00 0          
f701d000-f701e000 rw-p 00000000 00:00 0          
f701e000-f701f000 r--p 00000000 00:00 0          
f701f000-f7028000 r--p 00000000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libjava.so
f7028000-f703d000 r-xp 00009000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libjava.so
f703d000-f704b000 r--p 0001e000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libjava.so
f704b000-f704c000 r--p 0002b000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libjava.so
f704c000-f704d000 rw-p 0002b000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libjava.so
f704d000-f704f000 r--p 00000000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libverify.so
f704f000-f7057000 r-xp 00002000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libverify.so
f7057000-f705a000 r--p 0000a000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libverify.so
f705a000-f705b000 r--p 00000000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libverify.so
f705b000-f705c000 r--p 0000d000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libverify.so
f705c000-f705d000 rw-p 0000d000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/libverify.so
f705d000-f705f000 ---p 00000000 00:00 0          
f705f000-f7062000 ---p 00000000 00:00 0          
f7062000-f70b0000 rw-p 00000000 00:00 0          
f70b0000-f70b5000 rw-p 00000000 00:00 0          
f70b5000-f70b7000 r--p 00000000 00:00 0          /usr/lib/libgcc_s.so.1
f70b7000-f70cd000 r-xp 00002000 00:00 0          /usr/lib/libgcc_s.so.1
f70cd000-f70d1000 r--p 00018000 00:00 0          /usr/lib/libgcc_s.so.1
f70d1000-f70d2000 r--p 0001b000 00:00 0          /usr/lib/libgcc_s.so.1
f70d2000-f70d3000 rw-p 0001b000 00:00 0          /usr/lib/libgcc_s.so.1
f70d3000-f7162000 r--p 00000000 00:00 0          /usr/lib/libstdc++.so.6.0.28
f7162000-f720b000 r-xp 0008f000 00:00 0          /usr/lib/libstdc++.so.6.0.28
f720b000-f7263000 r--p 00138000 00:00 0          /usr/lib/libstdc++.so.6.0.28
f7263000-f7264000 r--p 00000000 00:00 0          /usr/lib/libstdc++.so.6.0.28
f7264000-f726c000 r--p 00190000 00:00 0          /usr/lib/libstdc++.so.6.0.28
f726c000-f726d000 rw-p 00190000 00:00 0          /usr/lib/libstdc++.so.6.0.28
f726d000-f726f000 rw-p 00000000 00:00 0          
f726f000-f7320000 r--p 00000000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/server/libjvm.so
f7320000-f777f000 r-xp 000b1000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/server/libjvm.so
f777f000-f7aab000 r--p 00510000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/server/libjvm.so
f7aab000-f7af9000 r--p 0083b000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/server/libjvm.so
f7af9000-f7b10000 rw-p 0083b000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/server/libjvm.so
f7b10000-f7f32000 rw-p 00000000 00:00 0          
f7f32000-f7f34000 r--p 00000000 00:00 0          /lib/libz.so.1.2.12
f7f34000-f7f43000 r-xp 00002000 00:00 0          /lib/libz.so.1.2.12
f7f43000-f7f4a000 r--p 00011000 00:00 0          /lib/libz.so.1.2.12
f7f4a000-f7f4b000 r--p 00017000 00:00 0          /lib/libz.so.1.2.12
f7f4b000-f7f4c000 rw-p 00017000 00:00 0          /lib/libz.so.1.2.12
f7f4c000-f7f4e000 r--p 00000000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/lib/i386/jli/libjli.so
f7f4e000-f7f57000 r-xp 00002000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/lib/i386/jli/libjli.so
f7f57000-f7f5d000 r--p 0000b000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/lib/i386/jli/libjli.so
f7f5d000-f7f5e000 r--p 00010000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/lib/i386/jli/libjli.so
f7f5e000-f7f5f000 rw-p 00010000 00:00 0          /usr/lib/jvm/java-1.8-openjdk/lib/i386/jli/libjli.so
f7f5f000-f7f63000 ---p 00000000 00:00 0          [vvar]
f7f63000-f7f65000 ---p 00000000 00:00 0          [vdso]
f7f65000-f7f77000 r--p 00000000 00:00 0          /lib/ld-musl-i386.so.1
f7f77000-f7fc6000 r--p 00012000 00:00 0          /lib/ld-musl-i386.so.1
f7fc6000-f7ffa000 r--p 00061000 00:00 0          /lib/ld-musl-i386.so.1
f7ffa000-f7ffb000 r--p 00094000 00:00 0          /lib/ld-musl-i386.so.1
f7ffb000-f7ffc000 rw-p 00094000 00:00 0          /lib/ld-musl-i386.so.1
f7ffc000-f7ffe000 rw-p 00000000 00:00 0          
fffaa000-fffab000 rwxp 00000000 00:00 0          
ffff8000-ffffe000 -w-p 00000000 00:00 0          [stack]

VM Arguments:
jvm_args: -Dapplication.home=/usr/lib/jvm/java-1.8-openjdk -Xms8m 
java_command: com.sun.tools.javac.Main -ms1g -mx1g HelloWorld.java
java_class_path (initial): /usr/lib/jvm/java-1.8-openjdk/lib/tools.jar:/usr/lib/jvm/java-1.8-openjdk/classes
Launcher Type: SUN_STANDARD

Environment Variables:
JAVA_HOME=/usr/lib/jvm/java-8-openjdk
PATH=/usr/lib/jvm/java-8-openjdk/bin:/sbin:/usr/sbin:/bin:/usr/bin:/usr/local/sbin:/usr/local/bin
LD_LIBRARY_PATH=/usr/lib/jvm/java-1.8-openjdk/jre/lib/i386/server:/usr/lib/jvm/java-1.8-openjdk/jre/lib/i386:/usr/lib/jvm/java-1.8-openjdk/jre/../lib/i386
SHELL=/bin/bash

Signal Handlers:
SIGSEGV: [libjvm.so+0x4fb6aa], sa_mask[0]=11111111111111111111111111111110, sa_flags=SA_RESTART|SA_SIGINFO
SIGBUS: [libjvm.so+0x4fb6aa], sa_mask[0]=11111111111111111111111111111110, sa_flags=SA_RESTART|SA_SIGINFO
SIGFPE: [libjvm.so+0x427b92], sa_mask[0]=11111111111111111111111111111110, sa_flags=SA_RESTART|SA_SIGINFO
SIGPIPE: [libjvm.so+0x427b92], sa_mask[0]=11111111111111111111111111111110, sa_flags=SA_RESTART|SA_SIGINFO
SIGXFSZ: [libjvm.so+0x427b92], sa_mask[0]=11111111111111111111111111111110, sa_flags=SA_RESTART|SA_SIGINFO
SIGILL: [libjvm.so+0x427b92], sa_mask[0]=11111111111111111111111111111110, sa_flags=SA_RESTART|SA_SIGINFO
SIGUSR1: SIG_DFL, sa_mask[0]=00000000000000000000000000000000, sa_flags=none
SIGUSR2: [libjvm.so+0x427ef6], sa_mask[0]=00000000000000000000000000000000, sa_flags=SA_RESTART|SA_SIGINFO
SIGHUP: [libjvm.so+0x42a419], sa_mask[0]=11111111111111111111111111111110, sa_flags=SA_RESTART|SA_SIGINFO
SIGINT: [libjvm.so+0x42a419], sa_mask[0]=11111111111111111111111111111110, sa_flags=SA_RESTART|SA_SIGINFO
SIGTERM: [libjvm.so+0x42a419], sa_mask[0]=11111111111111111111111111111110, sa_flags=SA_RESTART|SA_SIGINFO
SIGQUIT: [libjvm.so+0x42a419], sa_mask[0]=11111111111111111111111111111110, sa_flags=SA_RESTART|SA_SIGINFO


---------------  S Y S T E M  ---------------

OS:NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.14.3
PRETTY_NAME="Alpine Linux v3.14"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://bugs.alpinelinux.org/"

uname:Linux 4.20.69-ish SUPER AWESOME May 20 2023 23:41:32 i686
libc:glibc 2.9 NPTL 
rlimit: STACK 8192k, CORE 0k, NPROC 1024, NOFILE 4096, AS infinity
load average:1.17 1.18 1.15

/proc/meminfo:
MemTotal:        7896825 kB
MemFree:          943095 kB
MemShared:        943194 kB
Shmem:                 0 kB
Buffers:               0 kB
Cached:                0 kB
SwapTotal:             0 kB
SwapFree:              0 kB
Dirty:                 0 kB
Writeback:             0 kB
AnonPages:             0 kB
Mapped:                0 kB
Slab:                  0 kB


CPU:total 8 (initial active 8) (1 cores per cpu, 1 threads per core) family 0 model 0 stepping 0

/proc/cpuinfo:
processor	: 0
vendor_id	: iSH

processor	: 1
vendor_id	: iSH

processor	: 2
vendor_id	: iSH

processor	: 3
vendor_id	: iSH

processor	: 4
vendor_id	: iSH

processor	: 5
vendor_id	: iSH

processor	: 6
vendor_id	: iSH

processor	: 7
vendor_id	: iSH



Memory: 4k page, physical 0k(0k free), swap 0k(0k free)

vm_info: OpenJDK Server VM (25.302-b08) for linux-x86 JRE (1.8.0_302-b08), built on Dec  8 2021 12:27:28 by "buildozer" with gcc 10.3.1 20210424

time: Sat Sep 16 03:06:36 2023
timezone: UTC
elapsed time: 4.021455 seconds (0d 0h 0m 4s)


```

**Comment by schumacher-peter** [comment]
Thanks so much, you've saved me a lot of grief. BTW, in this line:

    java -ms128 -mx512m --version

you need to put an 'm' after the '8'.
